mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java

```diff
@@ -141,12 +142,25 @@ public ChangeEngineServiceImpl(@Autowired JobExecutionSourceChunkDao jobExecutio
         .map(parsedRecords))
       .onSuccess(parsedRecords -> {
         fillParsedRecordsWithAdditionalFields(parsedRecords);
-        boolean updateMarcActionExists = containsUpdateMarcActionProfile(jobExecution.getJobProfileSnapshotWrapper());
+
+        boolean updateMarcActionExists = containsMarcActionProfile(
+          jobExecution.getJobProfileSnapshotWrapper()
+          , List.of(FolioRecord.MARC_BIBLIOGRAPHIC, FolioRecord.MARC_AUTHORITY)
+          , Action.UPDATE);
+
+        boolean deleteMarcActionExists = containsMarcActionProfile(
+          jobExecution.getJobProfileSnapshotWrapper()
+          , List.of(FolioRecord.MARC_AUTHORITY)
+          , Action.DELETE);
 
         if (updateMarcActionExists) {
           updateRecords(parsedRecords, jobExecution, params)
             .onSuccess(ar -> promise.complete(parsedRecords))
             .onFailure(promise::fail);
+        } else if (deleteMarcActionExists) {
+          deleteRecords(parsedRecords, jobExecution, params)
+            .onSuccess(ar -> promise.complete(parsedRecords))
+            .onFailure(promise::fail);
```

Comment:
```suggestion

        if (updateMarcActionExists(jobExecution)) {
          updateRecords(parsedRecords, jobExecution, params)
            .onSuccess(ar -> promise.complete(parsedRecords))
            .onFailure(promise::fail);
        } else if (deleteMarcActionExists(jobExecution)) {
          deleteRecords(parsedRecords, jobExecution, params)
            .onSuccess(ar -> promise.complete(parsedRecords))
            .onFailure(promise::fail);
```
If you create 2 additional methods `updateMarcActionExists(jobExecution)` and `deleteMarcActionExists(jobExecution)` then amount of calls to `containsMarcActionProfile` method will be decreased.